apps/gallery/js/MetadataParser.js

```diff
@@ -500,60 +586,67 @@ var metadataParser = (function() {
         canvas.toBlob(function(blob) {
```

Comment:
On debugging, I realised without patch with big images the code was failing at context.drawImage call inside createThumbnailFromElement

https://github.com/mozilla-b2g/gaia/pull/33244/files#diff-f50b6edc68dfe30b26bf9d6977594e04R122

The only error seen in logs is  E/Gallery ( 7152): [JavaScript Error: "Image corrupt or truncated." {file: "blob:app://gallery.gaiamobile.org/deb74ba2-08ac-4c9c-80ae-b0a5877fb17e" line: 0}]

which is not very helpful, We are already using try catch but that fails to catch this error.